test/utils/utils.go

```diff
@@ -21,3 +31,683 @@ func GenerateAwsCredentialsSecretFromEnv(secretName, namespace string) *apiv1.Se
+							Operator: "Exists",
+						},
+					},
+					Containers: []apiv1.Container{
+						{
+							Name:  "etcd",
+							Image: "k8s.gcr.io/etcd:3.1.12",
+							VolumeMounts: []apiv1.VolumeMount{
+								{
+									Name:      "etcd-data-dir",
+									MountPath: "/etcd-data-dir",
+								},
+							},
+							Env: []apiv1.EnvVar{
+								{
+									Name:  "ETCD_DATA_DIR",
+									Value: "/etcd-data-dir",
+								},
+							},
+							Command: []string{
+								"/usr/local/bin/etcd",
+								"--listen-client-urls",
+								"http://0.0.0.0:2379",
+								"--advertise-client-urls",
+								"http://localhost:2379",
+							},
+							Ports: []apiv1.ContainerPort{
+								{
+									ContainerPort: 2379,
+								},
+							},
+							Resources: apiv1.ResourceRequirements{
+								Requests: apiv1.ResourceList{
+									"cpu":    resource.MustParse("100m"),
+									"memory": resource.MustParse("50Mi"),
+								},
+								Limits: apiv1.ResourceList{
+									"cpu":    resource.MustParse("200m"),
+									"memory": resource.MustParse("300Mi"),
+								},
+							},
+							ReadinessProbe: &apiv1.Probe{
+								Handler: apiv1.Handler{
+									HTTPGet: &apiv1.HTTPGetAction{
+										Port: intstr.FromInt(2379),
+										Path: "/health",
+									},
+								},
+								InitialDelaySeconds: 10,
+								TimeoutSeconds:      2,
+								PeriodSeconds:       10,
+								SuccessThreshold:    1,
+								FailureThreshold:    1,
+							},
+							LivenessProbe: &apiv1.Probe{
+								Handler: apiv1.Handler{
+									HTTPGet: &apiv1.HTTPGetAction{
+										Port: intstr.FromInt(2379),
+										Path: "/health",
+									},
+								},
+								InitialDelaySeconds: 10,
+								TimeoutSeconds:      2,
+								PeriodSeconds:       10,
+								SuccessThreshold:    1,
+								FailureThreshold:    3,
+							},
+						},
+					},
+					Volumes: []apiv1.Volume{
+						{
+							Name: "etcd-data-dir",
+							VolumeSource: apiv1.VolumeSource{
+								HostPath: &apiv1.HostPathVolumeSource{
+									Path: "/etc/kubernetes",
+									Type: &hostPathDirectoryOrCreate,
+								},
+							},
+						},
+					},
+					TerminationGracePeriodSeconds: &terminationGracePeriodSeconds,
+				},
+			},
+		},
+	}
+}
+
+func ClusterAPIEtcdService() *apiv1.Service {
+	return &apiv1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "etcd-clusterapi-svc",
+			Namespace: "default",
+			Labels: map[string]string{
+				"app": "etcd",
+			},
+		},
+		Spec: apiv1.ServiceSpec{
+			Ports: []apiv1.ServicePort{
+				{
+					Port:       2379,
+					TargetPort: intstr.FromInt(2379),
+					Name:       "etcd",
+				},
+			},
+			Selector: map[string]string{
+				"app": "etcd",
+			},
+		},
+	}
+}
+
+func TestingMachine(awsCredentialsSecretName string, clusterID string, namespace string) *clusterv1alpha1.Machine {
+	keyName := "libra"
+	publicIP := true
+	amiID := "ami-a9acbbd6"
+	iamInstanceProfileID := "openshift_master_launch_instances"
```

Comment:
can we move this to something known we precreate i.e "${var.enviroment_id}-worker-profile"